src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java

```diff
@@ -836,6 +890,428 @@ void should_insert_first_task_and_update_reconfiguration_task(String state, Stri
+                } else if ("matchNoRecordsByInvalidParams".equals(testName)) {
+                    callMarkReportTasksForRefresh(caseIds, taskIds, "WA",
+                                            "TESTAPPS", Collections.singletonList("DUMMY"), markBeforeTime);
+                } else {
+                    callMarkReportTasksForRefresh(caseIds, taskIds, "WA",
+                                                  "WAAPPS", taskStates, markBeforeTime);
+                }
+            }
+            default -> callMarkReportTasksForRefresh(caseIds, taskIds, "WA",
+                                                     "WAAPPS", taskStates, markBeforeTime);
+        }
+        taskRefreshTimestamps = callGetReportRefreshRequestTimes(taskIds);
+        Long count = taskRefreshTimestamps.stream().map(Objects::nonNull).count();
+        Assertions.assertEquals(expectedMarked, count, String.format("%s-%s:", testCategory, testName));
+    }
+
+    public Stream<Arguments> markTasksForRefreshArgsProvider() {
+        String waCaseId = "1690202048809201";
+        String terminatedStatus = "TERMINATED";
+        String waCaseTypeId = "WAAPPS";
+        String waJurisdiction = "WA";
+
+        String testCaseId = "16902020444444";
+        String unassignedStatus = "UNASSIGNED";
+        String testCaseTypeId = "TESTAPPS";
+        String testJurisdiction = "TEST";
+
+        String assignedStatus = "ASSIGNED";
+        String completedStatus = "COMPLETED";
+        String testCaseId1 = "1690202055555";
+
+        return Stream.of(
+            Arguments.arguments(
+                "ProcTestsWithDefaultNulls",
+                "matchAllRecords",
+                Stream.of(
+                    String.format("%s,%s,%s,%s,%s,%s", waCaseId, UUID.randomUUID(), terminatedStatus,
+                                  OffsetDateTime.now().minusDays(25), waCaseTypeId, waJurisdiction),
+                    String.format("%s,%s,%s,%s,%s,%s", waCaseId, UUID.randomUUID(), terminatedStatus,
+                                  OffsetDateTime.now().minusDays(24), waCaseTypeId, waJurisdiction)),
+                OffsetDateTime.now().minusDays(23).withHour(10).withMinute(0).withSecond(0).withNano(0),
+                2L),
+
+            Arguments.arguments(
+                "ProcTestsWithDefaultEmpty",
+                "matchAllRecords",
+                Stream.of(
+                    String.format("%s,%s,%s,%s,%s,%s", waCaseId, UUID.randomUUID(), terminatedStatus,
+                              OffsetDateTime.now().minusDays(27), waCaseTypeId, waJurisdiction),
+                    String.format("%s,%s,%s,%s,%s,%s", waCaseId, UUID.randomUUID(), terminatedStatus,
+                              OffsetDateTime.now().minusDays(28), waCaseTypeId, waJurisdiction)),
+                OffsetDateTime.now().minusDays(26).withHour(10).withMinute(0).withSecond(0).withNano(0),
+                2L),
+
+            Arguments.arguments(
```

Comment:
for the "MarkBeforeTimeTests" scenarios, a comment would help to clarify the condition for all, partial and no records refreshed

Reply:
Added a comment